text/0000-serialization.md

```diff
@@ -0,0 +1,187 @@
+      "content": 123
+    }
+  ]
+}
+```
+
+### Refracted Metadata & Attributes
+
+If the array's title from the example above needs to contain extra information, for example an encoding, then it can be refracted:
+
+```js
+{
+  "element": "array",
+  "meta": {
+    "title": {
+      "element": "string",
+      "attributes": {
+        "encoding": "utf-8"
+      },
+      "content": "My array"
+    }
+  },
+  "content": [
+    {
+      "element": "string",
+      "content": "Hello, world!"
+    },
+    {
+      "element": "number",
+      "content": 123
+    }
+  ]
+}
+```
+
+## Compact (list-based) JSON
+
+The short-form of the full JSON described above is based on lists. Each element is represented by a four item list in which the position of the item determines its semantic meaning. Each list consists of [`element`, `meta`, `attributes`, `content`].
+
+- `element`: The name of the element, e.g. `'string'`
+- `meta`: An object containing element metadata
+- `attributes`: An object containing element attributes
+- `content`: The element's content, which can be any type.
+
+### Example
+
+The following example describes an array named `My array` which looks like `['Hello, world!', 123]`:
+
+```js
+["array", {"title": "My array"}, {}, [
+  ["string", {}, {}, "Hello, world!"],
+  ["number", {}, {}, 123]
+]]
+```
+
+### Refracted Metadata & Attributes
+
+If the array's title from the example above needs to contain extra information, for example an encoding, then it can be refracted:
+
+```js
+["array", {
+    "title": ["string", {}, {"encoding": "utf-8"}, "My array"]
+  }, {}, [
+  ["string", {}, {}, "Hello, world!"],
+  ["number", {}, {}, 123]
+]]
+```
+
+## Preventing Ambiguity
+
+Because an element's metadata and attributes can contain both refracted elements and any other data, as well as fields which may be one or the other depending on content, we must prevent ambiguity. This can be accomplished by **always refracting ambiguous cases**. For example, given this **raw attribute** data:
+
+```js
+// Ambiguous raw content for full JSON serialization
+{
+  "element": "string",
+  "content": "foo"
+}
+
+// Ambiguous raw content for compact JSON serialization
+['foo', {}, {}, 0]
+```
+
+It must be serialized so that the parser knows it is **not an element**:
+
+```js
+// Unambiguous full JSON version
+{
+  "element": "object",
+  "content": [
+    {
+      "element": "member",
+      "content": {
+        "key": {
+          "element": "string",
+          "content": "element"
+        },
+        "value": {
+          "element": "string",
+          "content": "string"
+        }
+      }
+    },
+    {
+      "element": "member",
+      "content": {
+        "key": {
+          "element": "string",
+          "content": "content"
+        },
+        "value": {
+          "element": "string",
+          "content": "foo"
+        }
+      }
+    }
+  ]
+}
+
+// Unambiguous compact JSON version
+['array', {}, {}, ['foo', {}, {}, 0]]
```

Comment:
I would like to also explore how we could tell the parse that these are actual elements. We've talked about a few options here, but just wanted to make this comment. :)